### ISSUE TITLE
Issues with admin api with mTLS enabled

### ISSUE THREAD

**Issue by gugu** [redpanda-data/helm-charts#662]
### What happened?

When I enabled mTLS for API endpoint, I have these issues:
1. Readiness check does not work because it neither includes path to cert/key nor contains tls section of rpk. It can be solved by adding `config.rpk.tls` section to chart values
2. Upgrade script fails because it uses rpk without cert/key params and mounts emptydir as redpanda config. It can be solved only by disabling the upgrade script

Examples:

```
rpk cluster config import -f /tmp/base-config/bootstrap.yaml --api-urls example-1.example.default.svc.cluster.local.:9644,example-1.example.default.svc.cluster.local.:9644,example-2.example.default.svc.cluster.local.:9644 --admin-api-tls-enabled --admin-api-tls-truststore /etc/tls/certs/default/ca.crt
      rpk cluster config set storage_min_free_bytes 2684354560 --api-urls example-1.example.default.svc.cluster.local.:9644,example-1.example.default.svc.cluster.local.:9644,example-2.example.default.svc.cluster.local.:9644 --admin-api-tls-enabled --admin-api-tls-truststore /etc/tls/certs/default/ca.crt
```

### What did you expect to happen?

rpk-flags template needs to include flags for cert and key

### How can we reproduce it (as minimally and precisely as possible)?. Please include values file.

<details>

```console
$ helm get values <redpanda-release-name> -n <redpanda-release-namespace> --all
auth:
  sasl:
    enabled: true
    mechanism: SCRAM-SHA-512
    secretRef: redpanda-users
clusterDomain: cluster.local
commonLabels: {}
config:
  cluster: {}
  node: {}
  rpk:
    admin_api:
      tls:
        cert_file: /etc/tls/certs/default/tls.crt
        key_file: /etc/tls/certs/default/tls.key
        truststore_file: /etc/tls/certs/default/ca.crt
  tunable:
    compacted_log_segment_size: 67108864
    group_topic_partitions: 16
    kafka_batch_max_bytes: 1048576
    kafka_connection_rate_limit: 1000
    log_segment_size: 134217728
    log_segment_size_max: 268435456
    log_segment_size_min: 16777216
    max_compacted_log_segment_size: 536870912
    topic_partitions_per_shard: 1000
connectors:
  auth:
    sasl:
      enabled: false
      mechanism: scram-sha-512
      secretRef: ""
      userName: ""
  commonLabels: {}
  connectors:
    additionalConfiguration: ""
    bootstrapServers: ""
    brokerTLS:
      ca:
        secretNameOverwrite: ""
        secretRef: ""
      cert:
        secretNameOverwrite: ""
        secretRef: ""
      enabled: false
      key:
        secretNameOverwrite: ""
        secretRef: ""
    groupID: connectors-cluster
    producerBatchSize: 131072
    producerLingerMS: 1
    restPort: 8083
    schemaRegistryURL: ""
    secretManager:
      connectorsPrefix: ""
      consolePrefix: ""
      enabled: false
      region: ""
    storage:
      remote:
        read:
          config: false
          offset: false
          status: false
        write:
          config: false
          offset: false
          status: false
      replicationFactor:
        config: -1
        offset: -1
        status: -1
      topic:
        config: _internal_connectors_configs
        offset: _internal_connectors_offsets
        status: _internal_connectors_status
  container:
    javaGCLogEnabled: "false"
    resources:
      javaMaxHeapSize: 2G
      limits:
        cpu: 1
        memory: 2350Mi
      request:
        cpu: 1
        memory: 2350Mi
    securityContext:
      allowPrivilegeEscalation: false
  deployment:
    annotations: {}
    budget:
      maxUnavailable: 1
    create: false
    extraEnv: []
    livenessProbe:
      failureThreshold: 3
      initialDelaySeconds: 10
      periodSeconds: 10
      successThreshold: 1
      timeoutSeconds: 1
    nodeAffinity: {}
    nodeSelector: {}
    podAffinity: {}
    podAntiAffinity:
      custom: {}
      topologyKey: kubernetes.io/hostname
      type: hard
      weight: 100
    priorityClassName: ""
    progressDeadlineSeconds: 600
    readinessProbe:
      failureThreshold: 2
      initialDelaySeconds: 60
      periodSeconds: 10
      successThreshold: 3
      timeoutSeconds: 5
    restartPolicy: Always
    revisionHistoryLimit: 10
    schedulerName: ""
    securityContext:
      fsGroup: 101
      fsGroupChangePolicy: OnRootMismatch
      runAsUser: 101
    strategy:
      type: RollingUpdate
    terminationGracePeriodSeconds: 30
    tolerations: []
    topologySpreadConstraints:
    - maxSkew: 1
      topologyKey: topology.kubernetes.io/zone
      whenUnsatisfiable: ScheduleAnyway
    updateStrategy:
      type: RollingUpdate
  enabled: false
  fullnameOverride: ""
  global: {}
  image:
    pullPolicy: IfNotPresent
    repository: docker.redpanda.com/redpandadata/connectors
    tag: ""
  imagePullSecrets: []
  logging:
    level: warn
  monitoring:
    annotations: {}
    enabled: false
    labels: {}
    namespaceSelector:
      any: true
    scrapeInterval: 30s
  nameOverride: ""
  service:
    annotations: {}
    name: ""
    ports:
    - name: prometheus
      port: 9404
  serviceAccount:
    annotations: {}
    create: false
    name: ""
  storage:
    volume:
    - emptyDir:
        medium: Memory
        sizeLimit: 5Mi
      name: rp-connect-tmp
    volumeMounts:
    - mountPath: /tmp
      name: rp-connect-tmp
  test:
    create: false
  tolerations: []
console:
  affinity: {}
  annotations: {}
  autoscaling:
    enabled: false
    maxReplicas: 100
    minReplicas: 1
    targetCPUUtilizationPercentage: 80
  config: {}
  configmap:
    create: false
  console:
    config: {}
  deployment:
    create: false
  enabled: true
  extraContainers: []
  extraEnv: []
  extraEnvFrom: []
  extraVolumeMounts: []
  extraVolumes: []
  fullnameOverride: ""
  global: {}
  image:
    pullPolicy: IfNotPresent
    registry: docker.redpanda.com
    repository: redpandadata/console
    tag: ""
  imagePullSecrets: []
  ingress:
    annotations: {}
    className: ""
    enabled: false
    hosts:
    - host: chart-example.local
      paths:
      - path: /
        pathType: ImplementationSpecific
    tls: []
  livenessProbe:
    failureThreshold: 3
    initialDelaySeconds: 0
    periodSeconds: 10
    successThreshold: 1
    timeoutSeconds: 1
  nameOverride: ""
  nodeSelector: {}
  podAnnotations: {}
  podLabels: {}
  podSecurityContext:
    fsGroup: 99
    runAsUser: 99
  priorityClassName: ""
  readinessProbe:
    failureThreshold: 3
    initialDelaySeconds: 10
    periodSeconds: 10
    successThreshold: 1
    timeoutSeconds: 1
  replicaCount: 1
  resources: {}
  secret:
    create: false
    enterprise: {}
    kafka: {}
    login:
      github: {}
      google: {}
      jwtSecret: ""
      oidc: {}
      okta: {}
    redpanda:
      adminApi: {}
  secretMounts: []
  securityContext:
    runAsNonRoot: true
  service:
    annotations: {}
    port: 8080
    type: ClusterIP
  serviceAccount:
    annotations: {}
    create: true
    name: ""
  tolerations: []
  topologySpreadConstraints: {}
external:
  addresses:
  - redpanda-eu-central-1-0
  - redpanda-eu-central-1-1
  - redpanda-eu-central-1-2
  annotations:
    service.beta.kubernetes.io/aws-load-balancer-scheme: internal
    service.beta.kubernetes.io/aws-load-balancer-type: nlb-ip
  domain: example.com
  enabled: true
  externalDns:
    enabled: true
  type: LoadBalancer
fullnameOverride: ""
image:
  pullPolicy: IfNotPresent
  repository: docker.redpanda.com/redpandadata/redpanda
  tag: ""
imagePullSecrets: []
license_key: ""
license_secret_ref: {}
listeners:
  admin:
    external:
      default:
        advertisedPorts:
        - 31644
        enabled: false
        port: 9645
        tls:
          cert: external
    port: 9644
    tls:
      cert: default
      requireClientAuth: true
  http:
    authenticationMethod: null
    enabled: true
    external:
      default:
        advertisedPorts:
        - 30082
        authenticationMethod: null
        enabled: false
        port: 8083
        tls:
          cert: external
    kafkaEndpoint: default
    port: 8082
    tls:
      cert: default
      requireClientAuth: false
  kafka:
    authenticationMethod: null
    external:
      default:
        advertisedPorts:
        - 31093
        authenticationMethod: null
        port: 9094
        tls:
          cert: external
    port: 9093
    tls:
      cert: default
      requireClientAuth: false
  rpc:
    port: 33145
    tls:
      cert: default
      requireClientAuth: false
  schemaRegistry:
    authenticationMethod: null
    enabled: true
    external:
      default:
        advertisedPorts:
        - 30073
        authenticationMethod: null
        port: 8084
        tls:
          cert: external
    kafkaEndpoint: default
    port: 8081
    tls:
      cert: default
      requireClientAuth: false
logging:
  logLevel: info
  usageStats:
    enabled: true
monitoring:
  enabled: false
  labels: {}
  scrapeInterval: 30s
nameOverride: ""
node:
  developer_mode: false
nodeSelector: {}
post_install_job:
  enabled: true
post_upgrade_job:
  enabled: true
rackAwareness:
  enabled: false
  nodeAnnotation: topology.kubernetes.io/zone
rbac:
  annotations: {}
  enabled: false
resources:
  cpu:
    cores: 1
  memory:
    container:
      max: 2560Mi
      min: 2560Mi
serviceAccount:
  annotations: {}
  create: false
  name: ""
statefulset:
  additionalRedpandaCmdFlags: []
  annotations:
    prometheus.io/path: /public_metrics
    prometheus.io/port: "9644"
    prometheus.io/scheme: https
    prometheus.io/scrape: "true"
  budget:
    maxUnavailable: 1
  extraVolumeMounts: ""
  extraVolumes: ""
  initContainerImage:
    repository: busybox
    tag: latest
  initContainers:
    configurator:
      extraVolumeMounts: ""
      resources: {}
    extraInitContainers: ""
    setDataDirOwnership:
      enabled: false
      extraVolumeMounts: ""
      resources: {}
    setTieredStorageCacheDirOwnership:
      extraVolumeMounts: ""
      resources: {}
    tuning:
      extraVolumeMounts: ""
      resources: {}
  livenessProbe:
    failureThreshold: 3
    initialDelaySeconds: 10
    periodSeconds: 10
  nodeSelector: {}
  podAffinity: {}
  podAntiAffinity:
    custom: {}
    topologyKey: kubernetes.io/hostname
    type: hard
    weight: 100
  priorityClassName: ""
  readinessProbe:
    failureThreshold: 3
    initialDelaySeconds: 1
    periodSeconds: 10
    successThreshold: 1
  replicas: 3
  securityContext:
    fsGroup: 101
    fsGroupChangePolicy: OnRootMismatch
    runAsUser: 101
  sideCars:
    configWatcher:
      enabled: true
      extraVolumeMounts: ""
      resources: {}
      securityContext: {}
  startupProbe:
    failureThreshold: 120
    initialDelaySeconds: 1
    periodSeconds: 10
  terminationGracePeriodSeconds: 90
  tolerations: []
  topologySpreadConstraints:
  - maxSkew: 1
    topologyKey: topology.kubernetes.io/zone
    whenUnsatisfiable: ScheduleAnyway
  updateStrategy:
    type: RollingUpdate
storage:
  hostPath: ""
  persistentVolume:
    annotations: {}
    enabled: true
    labels: {}
    size: 50Gi
    storageClass: redpanda-xfs
  tieredConfig:
    cloud_storage_access_key: ""
    cloud_storage_api_endpoint: ""
    cloud_storage_azure_container: null
    cloud_storage_azure_shared_key: null
    cloud_storage_azure_storage_account: null
    cloud_storage_bucket: ""
    cloud_storage_cache_size: 5368709120
    cloud_storage_credentials_source: config_file
    cloud_storage_enable_remote_read: true
    cloud_storage_enable_remote_write: true
    cloud_storage_enabled: false
    cloud_storage_region: ""
    cloud_storage_secret_key: ""
  tieredStorageHostPath: ""
  tieredStoragePersistentVolume:
    annotations: {}
    enabled: false
    labels: {}
    storageClass: ""
tls:
  certs:
    default:
      caEnabled: true
    external:
      caEnabled: true
  enabled: true
tolerations: []
tuning:
  tune_aio_events: true
```

</details>

### Anything else we need to know?

I can provide PR

### Which are the affected charts?

Redpanda

### Chart Version(s)

<details>

```console
$ helm -n <redpanda-release-namespace> list 
redirector-redpanda   	default  	24      	2023-08-13 21:24:36.734329205 +0000 UTC	failed  	redpanda-5.1.2           	v23.2.4    

```

</details>

### Cloud provider

<details>
EKS
</details>

**Comment by joejulian** [comment]
I think will be solved by #658 do you agree?

**Comment by gugu** [comment]
I think they can be. is someone working on that issue? I can start

**Comment by joejulian** [comment]
@gugu Sorry, missed your reply last week. Nobody's working on it yet. It's likely to be started next week though.

If you do decide to work on #658 it would need to be gated on the Redpanda image tag version since we need to support at least two previous release versions.

**Comment by guicholeo** [comment]
I had the same issue with the communication between the redpandas using mTLS without the TLS enabled. I was able to fix it by adding the missing ports to the StatefulSet services and as well to the `redpanda` service. Had to modify the Helm Chart post render to include this ports since they are not enabled by default. Will be working on a PR to enable these ports if necessary.

**Comment by joejulian** [comment]
This should now be fixed with the release of redapanda chart 5.6.0. Can someone double check?

**Comment by alejandroEsc** [comment]
Closing this issue. If you feel this should not be the case, please update us @gugu and @guicholeo
